test/system/src/test/java/io/pravega/test/system/AbstractFailoverTests.java

```diff
@@ -0,0 +1,323 @@
+        assertTrue(deleteStreamStatus.get());
+        CompletableFuture<Boolean> deleteScopeStatus = controller.deleteScope(scope);
+        log.info("Deleting scope {}", scope);
+        assertTrue(deleteScopeStatus.get());
+    }
+
+    void createScopeAndStream(String scope, String stream, StreamConfiguration config, URI uri) {
+
+        try (StreamManager streamManager = new StreamManagerImpl(uri)) {
+            Boolean createScopeStatus = streamManager.createScope(scope);
+            log.info("Creating scope with scope name {}", scope);
+            log.debug("Create scope status {}", createScopeStatus);
+            Boolean createStreamStatus = streamManager.createStream(scope, stream, config);
+            log.debug("Create stream status {}", createStreamStatus);
+        }
+    }
+
+    void createReadersAndWriters(ClientFactory clientFactory, String readerGroupName, String scope,
+                                 ReaderGroupManager readerGroupManager, String stream, final int writers, final int readers) {
+        log.info("Client factory details {}", clientFactory.toString());
+        log.info("Creating {} writers", writers);
+        writerList = new ArrayList<>(writers);
+        log.info("Writers writing in the scope {}", scope);
+        EventStreamWriter<Long> tmpWriter;
+        for (int i = 0; i < writers; i++) {
+            log.info("Starting writer{}", i);
+            tmpWriter = clientFactory.createEventWriter(stream,
+                    new JavaSerializer<Long>(),
+                    EventWriterConfig.builder().retryAttempts(10).build());
+            writerList.add(tmpWriter);
+        }
+
+        log.info("Creating Reader group: {}, with readergroup manager using scope: {}", readerGroupName, scope);
+        readerGroupManager.createReaderGroup(readerGroupName, ReaderGroupConfig.builder().startingTime(0).build(),
+                Collections.singleton(stream));
+        log.info("Reader group name: {}, Reader group scope: {}, Online readers: {}",
+                readerGroupManager.getReaderGroup(readerGroupName).getGroupName(), readerGroupManager
+                        .getReaderGroup(readerGroupName).getScope(), readerGroupManager
+                        .getReaderGroup(readerGroupName).getOnlineReaders());
+        log.info("Creating {} readers", readers);
+        readerList = new ArrayList<>(readers);
+        log.info("Scope that is seen by readers {}", scope);
+        for (int i = 0; i < readers; i++) {
+            log.info("Starting reader: {}, with id: {}", i, readerName + i);
+            final EventStreamReader<Long> reader = clientFactory.createReader(readerName + i,
+                    readerGroupName,
+                    new JavaSerializer<Long>(),
+                    ReaderConfig.builder().build());
+            readerList.add(reader);
+        }
+
+        // start writing asynchronously
+        writerFutureList = writerList.stream().map(writer ->
```

Comment:
using createReaders and createWriters which use a completablefuture.runasync() for creatring writers and startwriting , creating readers and startreading.